google/cloud/storage/examples/storage_async_samples.cc

```diff
@@ -0,0 +1,250 @@
+  (client, argv.at(0), argv.at(1));
+}
+
+void DeleteObject(google::cloud::storage_experimental::AsyncClient& client,
+                  std::vector<std::string> const& argv) {
+  //! [delete-object]
+  namespace g = google::cloud;
+  namespace gcs_ex = google::cloud::storage_experimental;
+  [](gcs_ex::AsyncClient& client, std::string const& bucket_name,
+     std::string const& object_name) {
+    client.DeleteObject(bucket_name, object_name)
+        .then([](auto f) {
+          auto status = f.get();
+          if (!status.ok()) throw g::Status(std::move(status));
+          std::cout << "Object successfully deleted\n";
+        })
+        .get();
+  }
+  //! [delete-object]
+  (client, argv.at(0), argv.at(1));
+}
+
+void CreateClientCommand(std::vector<std::string> const& argv) {
+  if (!argv.empty()) throw examples::Usage("create-client");
+  CreateClient();
+}
+
+void CreateClientWithDPCommand(std::vector<std::string> const& argv) {
+  if (!argv.empty()) throw examples::Usage("client-client-with-dp");
+  CreateClientWithDP();
+}
+
+void AutoRun(std::vector<std::string> const& argv) {
+  if (!argv.empty()) throw examples::Usage{"auto"};
+  examples::CheckEnvironmentVariablesAreSet({
+      "GOOGLE_CLOUD_CPP_STORAGE_TEST_BUCKET_NAME",
+  });
+  auto bucket_name = google::cloud::internal::GetEnv(
+                         "GOOGLE_CLOUD_CPP_STORAGE_TEST_BUCKET_NAME")
+                         .value();
+  auto generator = google::cloud::internal::MakeDefaultPRNG();
+  auto const object_name = examples::MakeRandomObjectName(generator, "object-");
+
+  std::cout << "Running AsyncClient() example" << std::endl;
+  CreateClientCommand({});
+
+  std::cout << "Running AsyncClientWithDP() example" << std::endl;
+  CreateClientWithDPCommand({});
+
+  auto client = google::cloud::storage_experimental::MakeAsyncClient();
+
+  std::cout << "Running InsertObject() example" << std::endl;
+  InsertObject(client, {bucket_name, object_name});
+
+  std::cout << "Running InsertObjectVectorString() example" << std::endl;
+  InsertObjectVectorStrings(client, {bucket_name, object_name});
+
+  std::cout << "Running InsertObjectVector() example" << std::endl;
+  InsertObjectVector(client, {bucket_name, object_name});
+
+  std::cout << "Running InsertObjectVectorVector() example" << std::endl;
+  InsertObjectVectorVectors(client, {bucket_name, object_name});
+
+  std::cout << "Running DeleteObject() example" << std::endl;
+  DeleteObject(client, {bucket_name, object_name});
+}
+
+}  // namespace
+
+int main(int argc, char* argv[]) {
+  using Command =
+      std::function<void(google::cloud::storage_experimental::AsyncClient&,
+                         std::vector<std::string>)>;
+  auto make_entry =
+      [](std::string const& name, std::vector<std::string> arg_names,
+         Command const& command) -> examples::Commands::value_type {
+    arg_names.insert(arg_names.begin(), "<bucket-name>");
+    arg_names.insert(arg_names.begin(), "<object-name>");
```

Comment:
This will put `<object-name>` ahead of `<bucket-name>`, which is backwards.

Reply:
Argh, saw this too late.  Will send a PR in a minute.